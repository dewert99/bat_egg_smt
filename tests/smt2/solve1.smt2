(declare-fun x () Bool)
(declare-fun f (Bool) Bool)
(assert (and (= x (f x)) (f (f false))))
(check-sat-assuming ((not x)))
(get-unsat-core)
(check-sat)
(get-value (x (f true) (f false)))
(get-value ((f x) (f (f x))))