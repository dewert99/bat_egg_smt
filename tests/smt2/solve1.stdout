unsat
((not x))
sat
((x true)
 ((f true) true)
 ((f false) true))
(
 (define-fun f ((x!0 Bool)) Bool
  (ite (= x!0 false) true
  (ite (= x!0 true) true
   f!default)))
 (define-fun x () Bool true)
)
(((f x) true)
 ((f (f x)) true))
